### ISSUE TITLE
PR Downloader frequently fails to download latest package lists

### ISSUE THREAD

**Issue by ForbodingAngel** [spring/pr-downloader#119]
It tries, but they never actually get overwritten. I've never been able to nail down why it happens (it isn't a permissions issue, in my tests I have made the files/folders writable by anything). This was first an issue with SWL where people would have to manually delete rapid folders in order to get updates via the lobby.

I complained to many people, on the forums, etc and the bug still exists all these years later. For swl I simply started launching using a script that would delete the package lists and update things on it's own before running the lobby.

Now we are moving to chobby via gajop's launcher. Same issue is randomly happening. The thing that kills me is that this is honestly a stupid issue to have. PR Downloader should be manually deleting the lists once it's done with them anyway. There is no real value to having an old package list lying around.

**Comment by abma** [comment]
which version of pr-downloader is used?

also the console output when it failed please.

**Comment by ForbodingAngel** [comment]
Oh, never got the notification for this. RIP.

So I know why it's happening. I don't have console output because it's one of those things that happens commonly, but not to me. If you have any doubts about the frequency, just ask @active players in evo discord.

So why does it happen? Well, in the past it has been a permissions issue of some sort. My guess is not so much permissions but due to ownership. When steam installs a game, it's essentially just unpacking a zip file. It unpacks that zip file into a "common" folder that is writable by all (literally "everyone" on windows), although it has recently been changed to "users" (which means anyone with an account on the machine. This might mean integral accounts/groups like "TrustedInstaller", "Creator Owner", "System", etc. It just means that this account is authenticated at the very basic level.).

So basically, how is this PRD's problem? Honestly, it's PRD's problem because it doesn't delete it's package lists after using them. PRD has access to delete them, even if they existed before PRD or even if they were not created by PRD initially. PRD is being run as the current logged in user when launched.

The permissions setup for the common folder is like this:
https://i.imgur.com/m6ZBMd8.png

The two at the top are the more important ones ("User" group).

So how do I propose that this issue be fixed? Simple. Make PRD delete the rapid folder before it tries to get package lists and delete the rapid folder after it completes.

**Comment by abma** [comment]
i clearly don't understand your problem, sorry.

pr-downloader doesn't delete the package list because its used as cache. also its required for spring to make rapid:// dependencies work as they should.

**Comment by abma** [comment]
side note: this is a issue / bug tracker, not a tracker of solutions, thats what you wrote.

**Comment by ForbodingAngel** [comment]
Well, the solutions for users has always been to delete the rapid folder. You can solve it (or not), however you feel is best, but I see no harm in explaining to you how users have been solving it.

I understand that it is used as a cache, but PRD downloads the package lists every single time it is invoked. If this were about efficiency then that would be one facet, but obviously it isn't about efficiency. Therefore, I assume that function over form is appropriate in this case.

The problem is that the package lists do not get updated. The files do not get overwritten, appended to, updated, or any other more technically accurate term for it's process. The only way to get it to update properly is to delete the rapid folder. At which point it will generally work properly.

This doesn't happen with everyone, but it's been an issue for years. Up to this point I had simply written a script to bypass the issue, because I have reported this before and it still remains an issue. I complain about it, and get told to make an issue on github. Here we are. I'm sorry that I didn't reply to your initial request for feedback, but I never saw the email notifying me that you had responded, so I was unaware.

**Comment by ForbodingAngel** [comment]
Oh I never answered the version question. At the moment it is the one included with:
104.0.1-1344-g07bdbaf maintenance

**Comment by abma** [comment]
very likely you are hitting this bug:

https://github.com/spring/pr-downloader/commit/8eab580589f99b4252cf23ad9ae12c31f5cdf1ed

pr-downloader was updated there:

https://github.com/spring/pr-downloader/commit/8eab580589f99b4252cf23ad9ae12c31f5cdf1ed

so every build after 29. August should contain the fix.

**Comment by ForbodingAngel** [comment]
Oh excellent. Thanks for fixing that. It's been a thorn in our side for a long time. This will make life a lot easier :-)
